app/Models/Coupon.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Coupon extends Model
+{
+    protected $guarded = [];
+
+    public function user()
+    {
+        return $this->belongsTo(User::class, 'created_by');
+    }
+
+    public function redeem()
```

Comment:
is `redeem` a noun or a verb? does the method's functionality match its name?
for example, what do you think should be the difference in return types of `user->changeProfileImage()` and `user->profileImages()`